### PR TITLE
Achievements badge

### PR DESCRIPTION
**Pull request by prsauer**
Restore achievement lookup functionality

image

This is a rewrite from the old badge function and no longer relies on hard-coded achievement ids! It does however rely on some naming and conventions around the achievements. Could be issues in other languages?

### CODE REVIEW

File: packages/shared/src/api/blizzardApi.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+interface AuthResult {
+  access_token: string;
+  expires_in: string;
+  sub: string;
+  token_type: string;
+}
+
+async function refreshToken(region: string): Promise<string> {
+  const apiResult = await fetch(`https://${region}.battle.net/oauth/token?grant_type=client_credentials`, {
```
Comment (kunchenguid) [inline packages/shared/src/api/blizzardApi.ts:42]: i vaguely remember this should be region-less now. see https://oauth.battle.net/.well-known/openid-configuration. have you tried?

**Review by kunchenguid — APPROVED** [review]

File: packages/shared/src/components/CombatReport/CombatPlayers/CombatPlayer.tsx [hunk 2]
```diff
@@ -268,6 +269,7 @@ export function CombatPlayer(props: IProps) {
   return (
     <div className="flex flex-col flex-1 pb-4">
       <CombatUnitName unit={props.player} isTitle />
+      <AchievementBadge player={props.player} />
```
Comment (kunchenguid) [inline packages/shared/src/components/CombatReport/CombatPlayers/CombatPlayer.tsx:272]: might be good to wrap this with a mt-2

File: packages/shared/src/api/blizzardApi.ts (hunk 1 above)
Comment (prsauer) [reply packages/shared/src/api/blizzardApi.ts:42]: gonna take a look

Comment (prsauer) [reply packages/shared/src/api/blizzardApi.ts:42]: looks like only the oauth endpoint went regionless :\

**Pull request merged**